src/config.h.generic

```diff
@@ -58,7 +58,7 @@ sure both macros are undefined; an emulation function will then be used. */
 /* Define this if your compiler supports __attribute__((uninitialized)) */
 /* #undef HAVE_ATTRIBUTE_UNINITIALIZED */
 
-/* Define to 1 if you have the 'bcopy' function. */
+/* Define to 1 if you have the `bcopy' function. */
```

Comment:
Is this a new style?

Reply:
Interesting, yes, I noticed this too.

https://github.com/autotools-mirror/autoconf/commit/64df9b4523fe3b3c5d6e444fb65b1d92fc5073ce#diff-f3bb1e17a70d2e4733de29089bdd13e951406a1a15e4b6b0f5688668f5acbb2bR79

I found a change made a few years ago on the `‎lib/autoconf/functions.m4:80` file:

```diff
# Prepare the autoheader snippet for FUNCTION.
m4_define([_AH_CHECK_FUNC],
[AH_TEMPLATE(AS_TR_CPP([HAVE_$1]),
-  [Define to 1 if you have the `$1' function.])])
+  [Define to 1 if you have the '$1' function.])])

# _AC_CHECK_FUNCS_ONE_U(FUNCTION)
# -------------------------------
```

So apparently the `'straight'` quotes are the new style, in Autoconf 2.72 (shipped in Ubuntu 24.10 but not 24.04).

I honestly have no idea how the straight quotes came to be in the existing copy of config.h.generic. It's clearly a quite out-of-date file, not touched since the last PCRE2 release. I hypothesise that Philip perhaps had Autoconf 2.72 installed on his machine which last performed the update.

Let's just go with whatever style is produced by the version of Autoconf in Ubuntu 24.04, since that's what is most convenient to use in CI.

Reply:
I think it is because Philip runs with `LC_ALL=C` to avoid introducing "Unicode" quotes like this

Reply:
It's an ASCII `` ` `` backtick character.

I tried re-running `export LANG=C LC_ALL=C; ./autogen.sh; ./configure.sh; rm src/config.h.generic; make src/config.h.generic` all in a totally clean checkout. No change.

It really is due to the commit I linked above - the Autoconf behaviour has changed.

Philip uses Arch Linux I remember now. They updated to Autoconf 2.72, before Ubuntu did. So that's how Philip was using it this summer when he did the 10.44 release.